dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ZenodoHelper.java

```diff
@@ -33,48 +49,201 @@
+    private static String zenodoClientID;
+    private static String zenodoClientSecret;
 
     private ZenodoHelper() {
     }
 
+    public static void init(DockstoreWebserviceConfiguration configuration, HttpClient initHttpClient, SessionFactory initSessionFactory,
+            TokenDAO initTokenDAO, WorkflowDAO initWorkflowDAO, WorkflowVersionDAO initWorkflowVersionDAO) {
+        dockstoreUrl = configuration.getExternalConfig().computeBaseUrl();
+        dockstoreZenodoAccessToken = configuration.getDockstoreZenodoAccessToken();
+        dockstoreZenodoCommunityId = configuration.getDockstoreZenodoCommunityId();
+        httpClient = initHttpClient;
+        sessionFactory = initSessionFactory;
+        tokenDAO = initTokenDAO;
+        workflowDAO = initWorkflowDAO;
+        workflowVersionDAO = initWorkflowVersionDAO;
+        zenodoUrl = configuration.getZenodoUrl();
+        zenodoClientID = configuration.getZenodoClientID();
+        zenodoClientSecret = configuration.getZenodoClientSecret();
+
+        try {
+            dockstoreGA4GHBaseUrl = ToolsImplCommon.baseURL(configuration);
+        } catch (URISyntaxException e) {
+            LOG.error("Could create Dockstore base URL. Error is {}", e.getMessage(), e);
+            throw new CustomWebApplicationException("Could create Dockstore base URL. "
+                    + "Error is " + e.getMessage(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    /**
+     * Automatically registers the Zenodo DOI using the Dockstore Zenodo user.
+     * @param workflow
+     * @param workflowVersion
+     * @param workflowOwner
+     * @param authenticatedResourceInterface
+     * @return
+     */
+    public static void automaticallyRegisterDockstoreOwnedZenodoDOI(Workflow workflow, WorkflowVersion workflowVersion, User workflowOwner, AuthenticatedResourceInterface authenticatedResourceInterface) {
+        if (StringUtils.isEmpty(dockstoreZenodoAccessToken)) {
+            LOG.error("Dockstore Zenodo access token not found for automatic DOI creation, skipping");
+            return;
+        }
+
+        if (canAutomaticallyCreateDockstoreOwnedDoi(workflow, workflowVersion)) {
+            ApiClient zenodoClient = createDockstoreZenodoClient();
+            try {
+                // Perform some checks to increase the chance of a DOI being successfully created
+                checkCanRegisterDoi(workflow, workflowVersion, workflowOwner, DoiCreator.DOCKSTORE);
+                LOG.info("Automatically registering Dockstore owned Zenodo DOI for {}", workflowNameAndVersion(workflow, workflowVersion));
+                registerZenodoDOI(zenodoClient, workflow, workflowVersion, workflowOwner, authenticatedResourceInterface,
+                        DoiCreator.DOCKSTORE);
+            } catch (CustomWebApplicationException e) {
+                LOG.error("Could not automatically register DOI for {}", workflowNameAndVersion(workflow, workflowVersion), e);
```

Comment:
This is going to fail silently for the end user and we probably won't notice it either. I wonder if it's worth capturing this somehow?

* A CloudWatch Alarm based on this message?
* Add it as a warning to validations, if possible?

Reply:
I feel this could be a followup ticket; otherwise we won't know when/if things start failing (unless there's already another way we would notice).

Reply:
Created a follow up ticket https://ucsc-cgl.atlassian.net/browse/SEAB-6483